app/src/main/java/org/oppia/android/app/recyclerview/GridSpacingItemDecoration.kt

```diff
@@ -0,0 +1,135 @@
+      orientation = getOrientation(parent)
+    }
+    if (spanCount == -1) {
+      spanCount = getTotalSpan(parent)
+    }
+    val childIndex = parent.getChildAdapterPosition(view)
+    val itemSpanSize = getItemSpanSize(parent, childIndex)
+    val spanIndex = getItemSpanIndex(parent, childIndex)
+
+    val position = parent.getChildAdapterPosition(view)
+    val viewType = parent.adapter!!.getItemViewType(position)
+
+    if (viewType == HomeFragmentPresenter.ViewType.TOPIC_LIST.ordinal) {
+
+      if (spanCount < 1) return
+
+      if (spanCount == 2) {
+        outRect.right = halfSpacing
+        outRect.left = halfSpacing
+      }
+
+      if (spanCount == 4) {
+        outRect.left =
+          (spacing) - spanIndex * (spacing) / spanCount // spacing - column * ((1f / spanCount) * spacing)
+        outRect.right =
+          (spanIndex + 1) * (spacing) / spanCount // (column + 1) * ((1f / spanCount) * spacing)
+      } else {
+        // If it is in column 0 you apply the full offset on the start side, else only half
+        // If spanIndex + spanSize equals spanCount (it occupies the last column) you apply the full offset on the end, else only half.
+        when {
+          spanIndex == 0 -> {
+            outRect.left = spacing
+          }
+          itemSpanSize + spanIndex == spanCount -> {
+            outRect.right = spacing
+          }
+          else -> {
+            outRect.left = spacing / 2
+            outRect.right = spacing / 2
+          }
+        }
+      }
+
+      if (isLayoutRTL(parent)) {
+        val tmp = outRect.left
+        outRect.left = outRect.right
+        outRect.right = tmp
+      }
+
+      if (spanIndex < spanCount) {
+        outRect.top = halfSpacing
+      }
+      outRect.bottom = halfSpacing
+    }
+  }
+
+  private fun getTotalSpan(parent: RecyclerView): Int {
+    val layoutManager = parent.layoutManager
+    if (layoutManager is GridLayoutManager) {
+      return layoutManager.spanCount
+    }
+    return -1
+  }
+
+  private fun getItemSpanSize(parent: RecyclerView, childIndex: Int): Int {
+    val layoutManager = parent.layoutManager
+    if (layoutManager is GridLayoutManager) {
+      return layoutManager.spanSizeLookup.getSpanSize(childIndex)
+    }
+    return -1
+  }
+
+  private fun getItemSpanIndex(parent: RecyclerView, childIndex: Int): Int {
+    val layoutManager = parent.layoutManager
+    if (layoutManager is GridLayoutManager) {
+      return layoutManager.spanSizeLookup.getSpanIndex(childIndex, spanCount)
+    }
+    return -1
+  }
+
+  private fun getOrientation(parent: RecyclerView): Int {
+    val gridLayoutManager = parent.layoutManager
+    if (gridLayoutManager is GridLayoutManager) {
+      return gridLayoutManager.orientation
+    }
+    return VERTICAL
+  }
+
+  companion object {
+    private const val VERTICAL = OrientationHelper.VERTICAL
+
+    @SuppressLint("NewApi", "WrongConstant")
+    private fun isLayoutRTL(parent: RecyclerView): Boolean {
```

Comment:
Nit: ``isRtlLayout`` (switched the word order, and we should say 'Rtl' not 'RTL' per camel-casing guidelines in the style guide).

Reply:
Thanks! done.